docs/examples/helm-module/README.md

```diff
@@ -0,0 +1,124 @@
+
+- `.spec.definitionRef` is required to indicate the workload GVK in your Helm chart. For example, in our sample chart, the core workload is `deployments.apps/v1`.
+- `.spec.schematic.helm.template` is a string containing Helm release & repository definition in YAML format. 
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: WorkloadDefinition
+metadata:
+  name: webapp-chart
+  annotations:
+    definition.oam.dev/description: helm chart for webapp
+spec:
+  definitionRef:
+    name: deployments.apps
+    version: v1
+  schematic:
+    helm:
+      release:
+        chart:
+          spec:
+            chart: "podinfo"
+            version: "5.1.4"
+      repository:
+        url: "https://captainroy-hy.github.io/podinfo/"
+```
+Specifically, the definition follows the APIs from `fluxcd/flux2`, [HelmReleaseSpec](https://github.com/fluxcd/helm-controller/blob/main/docs/api/helmrelease.md) and [HelmRepositorySpec](https://github.com/fluxcd/source-controller/blob/main/docs/api/source.md#source.toolkit.fluxcd.io/v1beta1.HelmRepository).
+However, the fields shown in the sample are almost enough to describe a Helm chart release and its repository.
+
+## Define Application & Deploy
+
+Here is an example `Application`.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Application
+metadata:
+  name: myapp
+  namespace: default
+spec:
+  components:
+    - name: demo-podinfo 
+      type: webapp-chart # workload definiition
+      settings: # Helm chart values
+        replicaCount: 2
+```
+Diff from applications using CUE module workload, Helm module workload will use data in `settings` as the [Helm chart values](https://github.com/captainroy-hy/podinfo/blob/master/charts/podinfo/values.yaml).
+
+Now we can deploy the application.
+
+```shell
+kubectl apply -f webapp-chart-wd.yaml 
+
+kubectl apply -f myapp.yaml
+```
+
+After several minutes (it takes time to fetch Helm chart from the repo, render and install), you can check the Helm release is installed.
+```shell
+helm ls -A
+
+myapp-demo-podinfo	default  	1 	2021-03-05 02:02:18.692317102 +0000 UTC	deployed	podinfo-5.1.4   	5.1.4
+
+```
+And check the deployment defined in the chart.
+```shell
+kubectl get deploy
+
+NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
+myapp-demo-podinfo   1/1     1            1           66m
+```
+## Use existing Trait system
+
+A Helm module workload can fully work with Traits in the same way as existing workloads. For example
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Application
+metadata:
+  name: myapp
+  namespace: default
+spec:
+  components:
+    - name: demo-podinfo # helm chart release name
+      type: webapp-chart # definition = helm chart
+      settings: # helm chart values
+        replicaCount: 2
+      traits:
+        - name: scaler
+          properties:
+            replicas: 4 # trait will override the replicaCount in chart Values 
```

Comment:
I don't think "override" is a feature for traits. Actually, the `settings` section is already the override to default values of the chart.

Some useful traits could be label/annotation amender, patch trait (it will be interesting to see patch works with helm chart), node-affinity attacher etc.

Reply:
Got it. I will use image as a sample settings parameter. And add examples based on patch trait & label/annotation amender trait